scalafix/rules/src/main/scala/rsc/rules/RscCompat.scala

```diff
@@ -66,6 +75,17 @@ case class RscCompat(legacyIndex: SemanticdbIndex, config: RscCompatConfig)
           loop(env1, templ)
         case defn @ Defn.Class(_, name, _, _, templ) if defn.isVisible =>
           val env1 = TemplateScope(index.symbols, name.symbol.get.syntax) :: env
+
+          templ.inits.headOption.foreach { init =>
+            val tokens = init.tpe.tokens
+            // If type params of init are inferred
+            if (!tokens.exists {
+                  case Token.LeftBracket() => true
+                  case _ => false
+                }) {
```

Comment:
This check looks insufficient in situations like `class B; class C extends B`.

Reply:
That's true, we would generate a `RewriteInit` despite the parent having no type arguments. In the end, the patch would be empty though. We can avoid "no-op" `RewriteTarget`s in the collect but I kind of like the idea of letting the no-op happen in ascribe because it means `ascribeReturnType` (which should be renamed) does all the figuring-out of what the type should be, and these no-ops are just when the typestring is empty.

Reply:
I see. Then the "If type params of init are inferred" comment looks incorrect. Perhaps it could say "may be inferred"?

Reply:
I see. Then the "If type params of init are inferred" comment looks incorrect. Perhaps it could say "may be inferred"?